### PR TITLE
fix(store): eliminate SQLITE_BUSY contention during embedding writes (#161)

### LINKED ISSUE #161: [P3] store: investigate SQLite contention during embedding writes
## Goal
Eliminate (or quiet) the `database is locked (5) (SQLITE_BUSY)` warning that appears during embedding writes on first-time indexing.

## Motivation
Observed during a fresh index of ~89 PDFs (~2,700 chunks) on the BVI legal corpus. From the server log:

```
mark embedded attempt 1/3 failed: database is locked (5) (SQLITE_BUSY) labels=[1830 1831 ... 1849]
```

Indexing self-recovered (retry succeeded — indexed=91, embedded=2687, errors=0). So the contention is between concurrent writers within the indexing pipeline itself, likely between the chunk-write path and the embed-mark path.

Not user-visible failure, but:
- Noise pollutes operator logs.
- Larger corpora may hit the retry ceiling (3 attempts) and surface real errors.
- Suggests room to tighten transaction scoping or serialize the writers.

## Scope
- Audit write paths in `internal/store` and `internal/ingest`: identify which goroutines/transactions overlap.
- Consider one of:
  - Single writer goroutine with a channel-fed queue.
  - Wider PRAGMA `busy_timeout` / WAL tuning.
  - Restructuring the embedding-mark batch to coalesce into the chunk-write transaction.
- Add a regression test that runs a parallel index of N>1000 chunks and asserts zero busy retries.

## Acceptance Criteria
- No `SQLITE_BUSY` warnings during a full index of 5k+ chunks.
- No throughput regression on small corpora.

### PR DESCRIPTION
**Pull request by yidakra**
Closes #161.

## Summary
- `internal/store/sqlite_store.go`: extracted SQLite open into `openDB`, which calls `SetMaxOpenConns(1)` and sets `PRAGMA busy_timeout=5000` alongside the existing `journal_mode=WAL`.
- New `tests/store/sqlite_concurrent_writers_test.go` exercises 8 chunk writers + 4 embedding markers in parallel and asserts zero BUSY hits.

## Why
First-time indexing of moderately-sized corpora (~2.7k chunks) surfaced:
```
mark embedded attempt 1/3 failed: database is locked (5) (SQLITE_BUSY)
```
Indexing self-recovered on retry, but the warning indicated two writers contending. Root cause: `sql.Open` did not constrain the connection pool, so the pool could hand out multiple connections to concurrent goroutines (chunk writers and embedding-mark batches), each starting its own write transaction. SQLite allows only one concurrent writer.

`SetMaxOpenConns(1)` eliminates in-process contention by serializing writes through a single connection. WAL mode keeps reads non-blocking. `busy_timeout` guards against external concurrent processes.

## Test plan
- [x] `make ci` passes locally
- [x] New regression test passes under `-race`
- [ ] CI matrix on this PR

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **Bug Fixes**
  * Improved shutdown behavior to prevent spurious error messages during normal application termination.
  * Enhanced data storage reliability and performance under concurrent write operations through optimized connection settings and error handling.

* **Tests**
  * Added regression test to verify robust behavior when multiple write operations occur simultaneously.

### CODE REVIEW

File: internal/store/sqlite_store.go [hunk 1]
```diff
@@ -232,6 +232,35 @@ func NewSQLiteStore(path string) *SQLiteStore {
...
+func openDB(ctx context.Context, path string) (*sql.DB, error) {
+	db, err := sql.Open("sqlite", path)
+	if err != nil {
+		return nil, err
+	}
+	db.SetMaxOpenConns(1)
+	for _, pragma := range []string{
+		`PRAGMA journal_mode=WAL;`,
+		`PRAGMA busy_timeout=5000;`,
+	} {
```
[the remark this answers is not in the document]
Comment (yidakra) [reply internal/store/sqlite_store.go:261]: Addressed in d6e919a: PRAGMA order in openDB now applies busy_timeout BEFORE journal_mode=WAL so the WAL switch itself can wait under cross-process contention rather than returning BUSY immediately.
